modules/thermo-cycler/thermo-cycler-arduino/tc_timer.h

```diff
@@ -0,0 +1,25 @@
+#ifndef TC_TIMER_H_
+#define TC_TIMER_H_
+
+typedef enum{
+  IDLE,
+  RUNNING,
+  COMPLETE
+}Timer_status;
+
+class TC_Timer{
+  public:
+    unsigned int total_hold_time;
+    TC_Timer();
+    void reset();
+    bool start();
+    unsigned long time_left();
+    Timer_status get_status();
+    void update();
+  private:
+    unsigned long _total_hold_time_millis = 0;
```

Comment:
It's probably fine, but keep in mind that per [the arduino docs](https://www.arduino.cc/reference/en/language/functions/time/millis/) `millis` overflows after about 50 days, and institutions might keep their thermocycler plugged in all the time. It may be worth adding overflow handling.

Reply:
It's okay if the thermocycler is plugged in for more than 50 days because the timer will be running only in short durations while it's implementing cycles (which I doubt would be as long as 50 days). Here's a quick explanation on why this is overflow-safe: https://arduino.stackexchange.com/a/12591